cfme/tests/infrastructure/test_webmks_console.py

```diff
@@ -97,19 +120,45 @@ def test_webmks_vm_console(request, appliance, provider, vm_obj,
         # Enter Password:
         vm_console.send_keys("{}\n".format(console_vm_password))
 
-        result_after_login = ssh_client.run_command("who --count", ensure_user=True)
-        # Number of users before login would be 0 and after login would be 180
-        # If below assertion would fail result_after_login is also 0, denoting login failed
-        assert (result_before_login.output.split('=')[-1].strip() <
-             result_after_login.output.split('=')[-1].strip()), "Login Failed"
-
         logger.info("Wait to get the '$' prompt")
 
         vm_console.wait_for_text(text_to_find=provider.data.templates.get('console_template')
             ['prompt_text'], timeout=200)
 
+        def _validate_login():
+            # the following try/except is required to handle the exception thrown by SSH
+            # while connecting to VMware VM.It throws "[Error 104]Connection reset by Peer".
+            try:
+                result_after_login = ssh_client.run_command("who --count",
+                                            ensure_user=True)
+                logger.info("Output of 'who --count' is {} after login"
+                .format(result_after_login))
+                return result_before_login < result_after_login
+            except Exception as e:
```

Comment:
**REQUIRED**: If you know what exception can happen here (I presume it is an `IOError`) you should use that one here so you don't catch what you don't want to catch. `Exception` catches everything.